nifi-extension-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/proto/ProtoUtils.java

```diff
@@ -46,7 +48,21 @@ public static DynamicMessage createMessage(Descriptors.Descriptor descriptor, Ma
            switch (field.getType()) {
            case MESSAGE:
                if (field.isRepeated()) {
-                   Collection collection = value.getClass().isArray() ? Arrays.asList((Object[]) value) : (Collection) value;
+                   Collection collection = null;
+                   if (value.getClass().isArray()) {
+                       collection = Arrays.asList((Object[]) value);
```

Comment:
The original code casts all types of array into an `Object[]`. This only works for arrays with non-primitive items and thus the explicit `isArray` check can be replaced with a simpler (and more useful) `instanceof`.

This way we can use pattern-matching available in more modern Java and avoid an explicit additional cast.

```suggestion +1
                   if (value instanceof Object[] arrayValue) {
                       valueMaps = Arrays.stream(arrayValue)
                               .map(item -> (Map<String, Object>) item).toList();
```